src/Dependencies/Collections/ImmutableSegmentedDictionary`2+ValueBuilder.cs

```diff
@@ -0,0 +1,244 @@
+                        var valuesToAdd = ReadOnlyDictionary;
+                        _mutableDictionary = new SegmentedDictionary<TKey, TValue>(value);
+                        AddRange(valuesToAdd);
+                    }
+                }
+            }
+
+            public int Count => ReadOnlyDictionary.Count;
+
+            internal SegmentedDictionary<TKey, TValue> ReadOnlyDictionary => _mutableDictionary ?? _dictionary._dictionary;
+
+            IEnumerable<TKey> IReadOnlyDictionary<TKey, TValue>.Keys => throw new NotSupportedException();
+
+            IEnumerable<TValue> IReadOnlyDictionary<TKey, TValue>.Values => throw new NotSupportedException();
+
+            ICollection<TKey> IDictionary<TKey, TValue>.Keys => throw new NotSupportedException();
+
+            ICollection<TValue> IDictionary<TKey, TValue>.Values => throw new NotSupportedException();
+
+            bool ICollection<KeyValuePair<TKey, TValue>>.IsReadOnly => false;
+
+            ICollection IDictionary.Keys => throw new NotSupportedException();
+
+            ICollection IDictionary.Values => throw new NotSupportedException();
+
+            bool IDictionary.IsReadOnly => false;
+
+            bool IDictionary.IsFixedSize => false;
+
+            object ICollection.SyncRoot => throw new NotSupportedException();
+
+            bool ICollection.IsSynchronized => false;
+
+            public TValue this[TKey key]
+            {
+                get => ReadOnlyDictionary[key];
+                set => GetOrCreateMutableDictionary()[key] = value;
+            }
+
+            object? IDictionary.this[object key]
+            {
+                get => ((IDictionary)ReadOnlyDictionary)[key];
+                set => ((IDictionary)GetOrCreateMutableDictionary())[key] = value;
+            }
+
+            private SegmentedDictionary<TKey, TValue> GetOrCreateMutableDictionary()
+            {
+                return _mutableDictionary ??= new SegmentedDictionary<TKey, TValue>(_dictionary._dictionary, _dictionary.KeyComparer);
+            }
+
+            public void Add(TKey key, TValue value)
+            {
+                if (Contains(new KeyValuePair<TKey, TValue>(key, value)))
+                    return;
+
+                GetOrCreateMutableDictionary().Add(key, value);
+            }
+
+            public void Add(KeyValuePair<TKey, TValue> item)
+                => Add(item.Key, item.Value);
+
+            public void AddRange(IEnumerable<KeyValuePair<TKey, TValue>> items)
+            {
+                if (items == null)
+                    throw new ArgumentNullException(nameof(items));
+
+                foreach (var pair in items)
+                    Add(pair.Key, pair.Value);
+            }
+
+            public void Clear()
+            {
+                if (ReadOnlyDictionary.Count != 0)
+                {
+                    if (_mutableDictionary is null)
+                        _mutableDictionary = new SegmentedDictionary<TKey, TValue>(KeyComparer);
```

Comment:
>_mutableDictionary = new SegmentedDictionary<TKey, TValue>(KeyComparer); [](http://example.com/codeflow?start=24&length=72)

Could this set _dictionary to default similar to what the ImmutableSegmentedHashSet.ValueBuilder.Clear does?

Reply:
➡️ Updated to set `_dictionary` to default